File: js/clean.js
class ShapeBase {
  constructor() {}
  area() {
    return 0;
  }
}

class Square extends ShapeBase {
  side = 0;

  constructor(side) {
    super();
    this.side = side;
  }

  area() {
    return this.side * this.side;
  }
}

class Rectangle extends ShapeBase {
  width = 0;
  height = 0;

  constructor(width, height) {
    super();
    this.width = width;
    this.height = height;
  }

  area() {
    return this.width * this.height;
  }
}

class Triangle extends ShapeBase {
  base = 0;
  height = 0;

  constructor(base, height) {
    super();
    this.base = base;
    this.height = height;
  }

  area() {
    return 0.5 * this.base * this.height;
  }
}

class Circle extends ShapeBase {
  radius = 0;

  constructor(radius) {
    super();
    this.radius = radius;
  }

  area() {
    return Math.PI * this.radius * this.radius;
  }
}

function areaClean() {
  const shapes = new Array();
  for (let i = 0; i < 1000000; i++) {
    shapes.push(new Square(Math.random()));
    shapes.push(new Rectangle(Math.random(), Math.random()));
    shapes.push(new Triangle(Math.random(), Math.random()));
    shapes.push(new Circle(Math.random()));
  }

  let totalArea = 0;
  const start = new Date().valueOf();
  for (let i = 0; i < shapes.length; i++) {
    totalArea += shapes[i].area();
  }
  const end = new Date().valueOf();

  console.log("Processing time in ms =", end - start, " ms");
}

areaClean();

File: python/clean.py

from enum import Enum
from random import random
import math
import time



class Shape:
        def area(self):
                return 0
        
class Square(Shape) :
       def __init__(self,width):
              self.width=width
       def area(self):
              return self.width * self.width
              
class Rectangle(Shape) :
       def __init__(self,width,height):
              self.width=width
              self.height = height
       def area(self):
              return self.width * self.height
       
class Triangle(Shape) :
       def __init__(self,base,height):
              self.base=base
              self.height=height
       def area(self):
              return 0.5*self.base * self.height
       
class Circle(Shape) :
       def __init__(self,radius):
              self.radius=radius
       def area(self):
              return math.pi*self.radius*self.radius
       

shapes=[]

for i in range(1000000):
    shapes.append(Square(random()))
    shapes.append(Rectangle(random(),random()))
    shapes.append(Triangle(random(),random()))
    shapes.append(Circle(random()))


start = time.time()
totalArea = 0;
for shape in shapes:
    totalArea = totalArea + shape.area()
end = time.time()

print("Processing time in ms =", (end - start)*1000, " ms");


